pkg/gardenadm/botanist/kubelet.go

```diff
@@ -80,3 +84,38 @@ func (b *AutonomousBotanist) WriteKubeletBootstrapKubeconfig(ctx context.Context
 
 	return kubeletBootstrapKubeconfigCreator.Start(ctx)
 }
+
+// BootstrapKubelet bootstraps the kubelet.
+func (b *AutonomousBotanist) BootstrapKubelet(ctx context.Context) error {
+	node, err := nodeagent.FetchNodeByHostName(ctx, b.SeedClientSet.Client(), b.HostName)
+	if err != nil {
+		return fmt.Errorf("failed fetching node object via hostname: %w", err)
+	}
+	if node != nil {
+		b.Logger.Info("Found node for hostname, skipping kubelet bootstrap", "hostName", b.HostName, "nodeName", node.Name)
+		return nil
+	}
+
+	b.Logger.Info("No node found for hostname, bootstrapping kubelet", "hostName", b.HostName)
+
+	for _, obj := range nodeagentcomponent.GetCertificateSigningRequestClusterRoleBindings() {
+		desired := obj.DeepCopyObject()
+		if _, err := controllerutil.CreateOrUpdate(ctx, b.SeedClientSet.Client(), obj, func() error {
+			obj.(*rbacv1.ClusterRoleBinding).Subjects = desired.(*rbacv1.ClusterRoleBinding).Subjects
+			obj.(*rbacv1.ClusterRoleBinding).RoleRef = desired.(*rbacv1.ClusterRoleBinding).RoleRef
+			return nil
+		}); err != nil {
+			return fmt.Errorf("failed reconciling ClusterRoleBinding %q: %w", obj.GetName(), err)
+		}
+	}
```

Comment:
Why do we need to set `Subjects` and `RoleRef` again? It seems as if they would already be set.
What am I missing?

Reply:
`controllerutil.CreateOrUpdate` reads the `ClusterRoleBinding` from the system into `obj`. I.e, if the binding already exists, `obj` does not necessarily contain our desired `Subjects`/`RoleRef`. Hence we have to set them again.

Previously, we were using server-side apply here, but this is not supported by the fake client (for unit testing), so I decided to go back to this approach (also used commonly in g/g).